azurerm/resource_arm_monitor_diagnostics.go

```diff
@@ -0,0 +1,284 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/monitor/mgmt/2018-03-01/insights"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMonitorDiagnostics() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMonitorDiagnosticsCreateUpdate,
+		Read:   resourceArmMonitorDiagnosticsRead,
+		Update: resourceArmMonitorDiagnosticsCreateUpdate,
+		Delete: resourceArmMonitorDiagnosticsDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"target_resource_id": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
```

Comment:
can we remove the `DiffSuppressFunc` from this? in addition we can validate this using `ValidateFunc: azure.ValidateResourceID` (rather than the `azure.ValidateResourceIDOrEmpty` which is intended for optional fields)